### PR TITLE
Fix permission issues with `rsync -a`

### PR DESCRIPTION
**Pull request by anmolbabu**
Signed-off-by: anmolbabu <anmolbudugutta@gmail.com>

### CODE REVIEW

**Comment by cdrage** [comment]
Tests seem to be failing?

**Comment by kadel** [comment]
> Tests seem to be failing?

test is failing because this PR requires https://github.com/redhat-developer/odo/pull/1125 to work

**Comment by anmolbabu** [comment]
> > Tests seem to be failing?
> 
> test is failing because this PR requires [redhat-developer/odo#1125](https://github.com/redhat-developer/odo/pull/1125) to work

Not sure what that means @kadel 
Bcoz I see no issues except the ones with permissions in image PR which we resolved

**Comment by anmolbabu** [comment]
if [ ! -z "${ODO_S2I_WORKING_DIR}" ] && ([ "${ODO_S2I_SRC_BIN_PATH}" != "${ODO_S2I_WORKING_DIR}" ] && [ "${ODO_S2I_DEPLOYMENT_DIR}" != "${ODO_S2I_WORKING_DIR}" ])

**Comment by kadel** [comment]
we need `rsync -rlO` (recursive, links, omit dir times)

File: assemble-and-restart [hunk 1]
```diff
@@ -50,7 +50,7 @@ fi
 # We now run the assembly script. If there is a custom one written in the
 # source files, we use that instead.
 if [ -f ${ODO_S2I_SRC_BIN_PATH}/.s2i/bin/assemble ]; then
-    ${ODO_S2I_SRC_BIN_PATH}/.s2i/bin/assemble
+    ${ODO_S2I_SRC_BIN_PATH}/src/.s2i/bin/assemble
```
Comment (kadel) [inline assemble-and-restart:53]: this doesn't look right :-( we are testing different path than what we are executing `/src/`.
If `${ODO_S2I_SRC_BIN_PATH}/src/.s2i/bin/assemble` is correct path, it should be updated also in if condition.

**Review by kadel — CHANGES_REQUESTED** [review]

File: assemble-and-restart (hunk 1 above)
Comment (anmolbabu) [reply assemble-and-restart:53]: aah that's correct...
Fixing it

**Review by kadel — APPROVED** [review]

**Pull request merged**